numpy/core/records.py

```diff
@@ -423,6 +423,12 @@ def __new__(subtype, shape, dtype=None, buf=None, offset=0, strides=None,
                                       strides=strides, order=order)
         return self
 
+    def __array_finalize__(self, obj):
+        if type(obj) is not type(self):
+            # invoke __setattr__, which for void dtypes will do
+            # self.dtype = sb.dtype((record, self.dtype))
+            self.dtype = self.dtype
```

Comment:
OK, if you need it to be set also for explicit construction then, yes, your case is the right one. Since this was not obvious to me, and therefore may not be obvious to a future person looking at the code, maybe write the comment as:
```
# Both when constructing a new instance (obj=None) and when viewing
# from another class -- obj.view(type(self)) -- we invoke __setattr__,
# which for void dtypes will do self.dtype = sb.dtype((record, self.dtype)).
```